packages/next/src/server/app-render/app-render.tsx

```diff
@@ -947,6 +947,9 @@ async function renderToHTMLOrFlightImpl(
         renderServerInsertedHTML,
         serverCapturedErrors: allCapturedErrors,
         basePath: renderOpts.basePath,
+        traceData: renderOpts.experimental.clientTraceMetadata
```

Comment:
I put this feature behind an experimental flag for two reasons:
- This feature _is_ a bit experimental and maybe until we iron out some issues, it makes sense to put it behind an experiment.
- I wanted to make this feature opt-in because if people already have an OpenTelemetry propagator registered, the tracing data will go to the client and potentially leak data. It is unlikely, but it may happen.

If you feel like this is an acceptable risk, I am more than happy to remove the feature flag because it would drastically improve the out-of-the-box tracing experience with Next.js!